datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/types/dataset/DatasetType.java

```diff
@@ -127,8 +138,43 @@ public BrowseResults browse(@Nonnull List<String> path,
     }
 
     @Override
-    public List<BrowsePath> browsePaths(@Nonnull String urn) throws Exception {
+    public List<BrowsePath> browsePaths(@Nonnull String urn, @Nonnull final QueryContext context) throws Exception {
         final StringArray result = _datasetsClient.getBrowsePaths(DatasetUtils.getDatasetUrn(urn));
         return BrowsePathsMapper.map(result);
     }
+
+    @Override
+    public Class<DatasetUpdateInput> inputClass() {
+        return DatasetUpdateInput.class;
+    }
+
+    @Override
+    public Dataset update(@Nonnull DatasetUpdateInput input, @Nonnull QueryContext context) throws Exception {
+        // TODO: Verify that updater is owner.
+        final CorpuserUrn actor = new CorpuserUrn(context.getActor());
```

Comment:
should there be a flag for whether a dataset can only be edited by its owner? this sounds like a limiting feature most users wouldn't want by default

Reply:
Overall we need a more comprehensive story about roles and permissions in DH. You're right that it may not be required that we strictly limit to owners. Let's think more on this before I address that TODO^^